### PR TITLE
Add inference engine for segmenting texts in documents using Yolo

### PR DESCRIPTION
**Pull request by anuprulez**
The PR adds an inference engine for segmenting text blocks in documents using pretrained Yolo model: https://github.com/opendatalab/DocLayout-YOLO

Model: https://huggingface.co/juliozhao/DocLayout-YOLO-DocLayNet-Docsynth300K_pretrained

It creates bounding boxes around the recognized text and also provides the coordinates of the all the bounding boxes in a geojson format.

Docker container: https://github.com/usegalaxy-eu/docker-doclayout-yolo

Galaxy7- Segmented image

Galaxy9- Segmented image

ping @bgruening @Sch-Da

### CODE REVIEW

File: tools/image_processing/yolo/doclayoutyolo.xml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            --input_image_size '$input_image_size'
+            --output_image '$output_image'
+            --output_geojson '$output_segmentation_coordinates'
+]]>
+    </command>
+    <inputs>
+	    <param name="input_yolo_model" type="data" format="zip" label="Yolo model" help="Please upload a Yolo model."/>
+        <param name="input_image" type="data" format="tiff,jpg,png" label="Input image" help="Please provide an input image for the analysis."/>
+        <param name="input_confidence" type="float" label="Confidence" value="0.5" min="0.0" max="1.0" help="Set confidence threshold between 0.0 and 1.0 drawing bounding boxes"/>
+        <param name="input_image_size" type="integer" label="Image size" value="1024" min="1" max="1500" help="Set input image size for image resize by Doclayout Yolo model"/>
```
Comment (bgruening) [inline tools/image_processing/yolo/doclayoutyolo.xml]: this is always needed? So it will always resize the image?

File: tools/image_processing/yolo/doclayoutyolo.xml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    </outputs>
+    <tests>
+        <test>
+            <param name="input_yolo_model" value="input_yolo_model.zip" location="https://huggingface.co/juliozhao/DocLayout-YOLO-DocLayNet-Docsynth300K_pretrained/resolve/main/doclayout_yolo_doclaynet_imgsz1120_docsynth_pretrain.pt?download=true"/>
+            <param name="input_image" value="input_image_png.png"/>
+            <param name="input_confidence" value="0.5"/>
+            <param name="input_image_size" value="1024"/>
+            <output name="output_image" ftype="png">
+                <assert_contents>
+                    <has_size size="920950" delta="100" />
```
Comment (bgruening) [inline tools/image_processing/yolo/doclayoutyolo.xml]: Please also try to use the new image assertions. (even if the linting will fail then)

File: tools/image_processing/yolo/doclayoutyolo.xml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    <description>through Diverse Synthetic Data and Global-to-Local Adaptive Perception</description>
+    <macros>
+        <token name="@TOOL_VERSION@">0.0.4.1</token>
+        <token name="@VERSION_SUFFIX@">0</token>
+        <token name="@PROFILE@">23.2</token>
+    </macros>
+    <creator>
+        <organization name="European Galaxy Team" url="https://galaxyproject.org/eu/"/>
+        <person givenName="Anup" familyName="Kumar" email="kumara@informatik.uni-freiburg.de"/>
+    </creator>
```
Comment (bgruening) [inline tools/image_processing/yolo/doclayoutyolo.xml]: Can you please add main.py (maybe rename it to something more specific) as https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-required-files

File: tools/image_processing/yolo/doclayoutyolo.xml [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+            --input_image '$input_image'
+            --input_image_ext '$input_image.ext'
+            --input_confidence '$input_confidence'
+            --input_image_size '$input_image_size'
+            --output_image '$output_image'
+            --output_geojson '$output_segmentation_coordinates'
+]]>
+    </command>
+    <inputs>
+	    <param name="input_yolo_model" type="data" format="zip" label="Yolo model" help="Please upload a Yolo model."/>
```
Comment (bgruening) [inline tools/image_processing/yolo/doclayoutyolo.xml]: something off with inedentation here?

File: tools/image_processing/yolo/test-data/input_image_jpg.jpg
Comment (bgruening) [inline tools/image_processing/yolo/test-data/input_image_jpg.jpg:1]: can we store that on zenodo? It too large ...

File: tools/image_processing/yolo/doclayoutyolo.xml (hunk 1 above)
Comment (anuprulez) [reply tools/image_processing/yolo/doclayoutyolo.xml]: this is a hyperparameter: lower size will work faster with lower accuracy, larger size will be slower with higher accuracy. Balanced value is 1024

**Comment by Sch-Da** [comment]
Looking really good, thank you @anuprulez!

File: tools/image_processing/yolo/doclayoutyolo.xml (hunk 1 above)
Comment (bgruening) [reply tools/image_processing/yolo/doclayoutyolo.xml]: I think this sentence should go to the help text :)

File: tools/image_processing/yolo/doclayoutyolo.xml [hunk 5]
```diff
@@ -0,0 +1,124 @@
+<tool id="doclayout_yolo" name="DocLayout-YOLO: Enhancing Document Layout Analysis" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="@PROFILE@">
+    <description>through Diverse Synthetic Data and Global-to-Local Adaptive Perception</description>
```
Comment (bgruening) [inline tools/image_processing/yolo/doclayoutyolo.xml]: Isn't that description too detailed. Maybe the name simply "doclayout-yolo" and the description "enhancing document layout analysis"? The current description maybe move to the help?

File: tools/image_processing/yolo/doclayoutyolo.xml (hunk 1 above)
Comment (anuprulez) [reply tools/image_processing/yolo/doclayoutyolo.xml]: Added more helpful descriptions, also for `confidence` parameter

File: tools/image_processing/yolo/doclayoutyolo.xml (hunk 5 above)
Comment (anuprulez) [reply tools/image_processing/yolo/doclayoutyolo.xml]: Ok, done!

File: tools/image_processing/yolo/doclayoutyolo.xml (hunk 3 above)
Comment (anuprulez) [reply tools/image_processing/yolo/doclayoutyolo.xml]: Done!

File: tools/image_processing/yolo/doclayoutyolo.xml (hunk 2 above)
Comment (anuprulez) [reply tools/image_processing/yolo/doclayoutyolo.xml]: do you mean these types of assertions for image: https://docs.galaxyproject.org/en/latest/dev/schema.html#has-image-center-of-mass ? There are a lot many such as https://docs.galaxyproject.org/en/latest/dev/schema.html#has-image-mean-object-size? 

thanks!

Comment (anuprulez) [reply tools/image_processing/yolo/doclayoutyolo.xml]: I added a few of these asserts for output images

```
<has_image_channels channels="3" />
<has_image_height height="2721" />
<has_image_width width="1700" />
```

But, the tool tests fail for TIFF output image with:

`<COMPRESSION.LZW: 5> requires the 'imagecodecs' package`

File: tools/image_processing/yolo/.shed.yml [hunk 6]
```diff
@@ -1,13 +1,17 @@
...
-type:
+homepage_url: https://github.com/bgruening/galaxytools
 categories:
-  - Machine Learning
   - Imaging
-maintainers:
-  - anuprulez
+description: tools for building yolo workflows.
+long_description:   a collection of tools for training yolo model.
+remote_repository_url: https://github.com/bgruening/galaxytools/tree/master/tools
```
Comment (bgruening) [inline tools/image_processing/yolo/.shed.yml]: this is not correct

**Review by bgruening — APPROVED** [review]
